### PR TITLE
[MLP-148] Add --no-api option to skip API files

### PR DESCRIPTION
**Pull request by matiasbeckerle**
## Background

@juampick came with the idea of providing some optional parameters and one is `--no-api` to avoid including all the backend files because sometimes we rely on other APIs.

## Changes

- Added `--no-api` option.
- Conditionally add API files based on option provided.

## Notes

There was also another idea, about providing `--create`, `--read`, `--update` and `--delete` options but while I was doing that I've found that the template and generator code was started to look much more complex and way too difficult to read & maintain. I published another WIP branch if someone wants to resume that work but I discourage to do so (really, not nice). Here is the branch: https://github.com/matiasbeckerle/mern-seed/tree/MLP-148-crud-parameters-wip

### CODE REVIEW

File: cli/generators/crud/index.js [hunk 1]
```diff
@@ -7,6 +7,13 @@ module.exports = class extends Generator {
     super(args, opts);
 
     this.argument('name', { type: String, required: true });
+
+    this.option('api', {
```
Comment (AlphaGit) [inline cli/generators/crud/index.js]: Here the option is `api`, but the docs say `--no-api`. Maybe this one should be changed?

**Review by AlphaGit — CHANGES_REQUESTED** [review]

File: cli/generators/crud/index.js (hunk 1 above)
Comment (matiasbeckerle) [inline cli/generators/crud/index.js]: No. The `no-` prefix is like the negative version of your option. I didn't invest time on researching it but I believe is on purpose so you can use `--api` or `--no-api`. Actually I first tried with `no-api` but it doesn't work. I believe that if I dig a little bit on the documentation I will find the explanation...

Comment (AlphaGit) [inline cli/generators/crud/index.js]: Wonderful. I didn't know about that. Thanks!

**Review by AlphaGit — APPROVED** [review]

**Review by juampick — APPROVED** [review]
Thanks Matts @matiasbeckerle Great job! really neat

**Comment by matiasbeckerle** [comment]
Appreciated @juampick!

**Pull request merged**